pkg/balancer-js/src/utils/signatures.ts

```diff
@@ -0,0 +1,143 @@
+import { MaxUint256 as MAX_DEADLINE } from '@ethersproject/constants';
+import { Contract } from '@ethersproject/contracts';
+import { hexValue, hexZeroPad, splitSignature } from '@ethersproject/bytes';
+import { BigNumberish } from '@ethersproject/bignumber';
+import { Signer, TypedDataSigner } from '@ethersproject/abstract-signer';
+
+export enum RelayerAction {
+  JoinPool = 'JoinPool',
+  ExitPool = 'ExitPool',
+  Swap = 'Swap',
+  BatchSwap = 'BatchSwap',
+  SetRelayerApproval = 'SetRelayerApproval',
+}
+
+export type Account = string | Signer | Contract;
+
+export async function accountToAddress(account: Account): Promise<string> {
+  if (typeof account == 'string') return account;
+  if (Signer.isSigner(account)) return account.getAddress();
+  if (account.address) return account.address;
```

Comment:
```suggestion
  if (account.address) return account.address;
  if (Signer.isSigner(account)) return account.getAddress();
```

Subtle, but this will save RPCs when using SignerWithAddress

Reply:
This would need some nasty type assertions to work and calling `getAddress` on a `SignerWithAddress` just returns `this.address` so the current code is equivalent.